nats.go

```diff
@@ -1028,10 +1028,13 @@ func (nc *Conn) deliverMsgs(s *Subscription) {
 		s.mu.Lock()
 		max = s.max
 		closed = s.closed
-		s.delivered++
-		delivered = s.delivered
 		s.mu.Unlock()
 
+		// Use the same semantics everywhere for delivered
```

Comment:
I think it was fine leaving delivered above. In case on async subscription, this variable is updated only here. It does not even need any locking, but clearly no need for atomic.